src/gui/programupdater.h

```diff
@@ -57,14 +58,22 @@ class ProgramUpdater final : public QObject
 
 private slots:
     void rssDownloadFinished(const Net::DownloadResult &result);
-    void fallbackDownloadFinished(const Net::DownloadResult &result);
+    void fallbackDownloadFinished(const Net::DownloadResult &result, Version &version);
 
 private:
+    enum class RemoteSources
+    {
+        Fosshub,
+        QbtMain,
+        QbtBackup
+    };
+
     void handleFinishedRequest();
-    bool shouldUseFallback() const;
+    RemoteSources selectRemote() const;
```

Comment:
```suggestion
    RemoteSource selectRemoteSource() const;
```

Reply:
I would even call it `getBestRemoteSource()`. For me, "select" is associated with producing side effect.